src/groth16/aggregate/srs.rs

```diff
@@ -115,3 +321,32 @@ fn structured_generators_scalar_power<G: CurveProjective>(
     );
     powers_of_g.into_iter().map(|v| v.into_affine()).collect()
 }
+
+fn write_vec<G: CurveAffine, W: Write>(w: &mut W, v: &[G]) -> io::Result<()> {
+    w.write_u32::<BigEndian>(v.len() as u32)?;
+    for p in v {
+        write_point(w, p)?;
+    }
+    Ok(())
+}
+
+fn write_point<G: CurveAffine, W: Write>(w: &mut W, p: &G) -> io::Result<()> {
+    w.write_all(p.into_compressed().as_ref())?;
+    Ok(())
+}
+
+fn read_vec<G: CurveAffine, R: Read>(r: &mut R) -> io::Result<Vec<G>> {
+    let vector_len = r.read_u32::<BigEndian>()? as usize;
+    let mut data = vec![G::Compressed::empty(); vector_len];
+    for encoded in &mut data {
+        r.read_exact(encoded.as_mut())?;
+    }
+    Ok(data
+        .par_iter()
+        .map(|enc| {
+            enc.into_affine()
+                .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))
+                .and_then(|s| Ok(s))
+        })
+        .collect::<io::Result<Vec<_>>>()?)
+}
```

Comment:
should have a unit tests for the reading and writing methods

Reply:
@cryptonemo made a test already in `tests/groth16_aggregation.rs`